ext/dom/document.c

```diff
@@ -878,6 +878,10 @@ PHP_METHOD(DOMDocument, createElementNS)
 
 	if (errorcode == 0) {
 		if (xmlValidateName((xmlChar *) localname, 0) == 0) {
+			/* https://dom.spec.whatwg.org/#validate-and-extract: requires the empty string uri to be set to NULL */
+			if (uri_len == 0) {
+				uri = NULL;
+			}
```

Comment:
It may make sense to start emitting a warning for empty strings, stating it should be non-empty or ``null`` so that we can elevate this to a ValueError in the next major.

Similarly with other such checks.

Reply:
I don't think we should emit a warning. The spec allows both NULL and the empty string, so we should just follow that. It's just that the spec treats them as equal by enforcing that the empty string is always set to NULL.

Reply:
Right, so it's just an internal issue. Your comment on the issue had me confused.